cpp/models/lct_secir/model.h

```diff
@@ -35,233 +35,372 @@ namespace mio
+            compress_vector(subcompartments_ts[timepoint], compartments);
+            compartments_ts.add_time_point(subcompartments_ts.get_time(timepoint), compartments);
+        }
+
+        return compartments_ts;
+    }
+
+    /**
+     * @brief Checks that the model satisfies all constraints (e.g. parameter or population constraints).
+     * @return Returns true if one or more constraints are not satisfied, false otherwise.
+     */
+    bool check_constraints() const
+    {
 
-        auto params           = this->parameters;
-        auto total_population = pop.sum() - pop[LctState::template get_first_index<InfectionState::Dead>()];
+        return (check_constraints_impl() || this->parameters.check_constraints() ||
+                this->populations.check_constraints());
+    }
+
+private:
+    /**
+     * @brief Converts a vector with subcompartments in a vector without subcompartments 
+     * by summing up subcompartment values.
+     * This is done recusively for each group which corresponds to a slice of the vector.
+     *
+     * @tparam group The group specifying the slice of the vector being considered. 
+     * @param[in] subcompartments The vector that should be converted.
+     * @param[out] compartments Reference to the vector where the output is stored.
+     */
+    template <size_t Group = 0, std::enable_if_t<(Group < m_groups) && (Group >= 0), bool> = true>
+    void compress_vector(const Eigen::VectorXd& subcompartments, Eigen::VectorXd& compartments) const
+    {
+        using LctState = typename std::tuple_element_t<Group, tupleLctStates>;
 
-        ScalarType infectedNoSymptoms = 0;
-        ScalarType infectedSymptoms   = 0;
-        ScalarType flow               = 0;
+        // Specify first indices of the vector slices considered.
+        Eigen::Index first_index_subcomps = this->populations.template get_first_index_group<Group>();
+        Eigen::Index count_InfStates      = (Eigen::Index)InfectionState::Count;
+        Eigen::Index first_index_comps    = Group * count_InfStates;
 
-        // Calculate sum of all subcompartments for InfectedNoSymptoms.
-        infectedNoSymptoms =
-            pop.segment(LctState::template get_first_index<InfectionState::InfectedNoSymptoms>(),
-                        LctState::template get_num_subcompartments<InfectionState::InfectedNoSymptoms>())
+        // Use segment of the vector subcompartments of each InfectionState and sum up the values of subcompartments.
+        compartments[first_index_comps + (Eigen::Index)InfectionState::Susceptible] =
+            subcompartments[first_index_subcomps];
+        compartments[first_index_comps + (Eigen::Index)InfectionState::Exposed] =
+            subcompartments
+                .segment(
+                    first_index_subcomps +
+                        std::tuple_element_t<Group,
+                                             tupleLctStates>::template get_first_index<InfectionState::Exposed>(),
```

Comment:
```suggestion
                                             tupleLctStates>::template get_first_index_state<InfectionState::Exposed>(),
```
For me this would be easier to understand to not get confused with all the indices.

Reply:
As this function is called very often, i would prefer to keep the name short. I think the meaning is clear because of the use of an InfectionState as a template parameter.